examples/tpm2-ekcert/main.go

```diff
@@ -38,19 +42,70 @@ func main() {
 	}
 }
 
-func readEKCert(path string, idx uint32) ([]byte, error) {
+func readEKCert(path string, certIdx, tmplIdx uint32) ([]byte, error) {
 	rwc, err := tpm2.OpenTPM(path)
 	if err != nil {
 		return nil, fmt.Errorf("can't open TPM at %q: %v", path, err)
 	}
 	defer rwc.Close()
-	ekCert, err := tpm2.NVRead(rwc, tpmutil.Handle(idx))
+	ekCert, err := tpm2.NVRead(rwc, tpmutil.Handle(certIdx))
 	if err != nil {
 		return nil, fmt.Errorf("reading EK cert: %v", err)
 	}
 	// Sanity-check that this is a valid certificate.
-	if _, err := x509.ParseCertificate(ekCert); err != nil {
+	cert, err := x509.ParseCertificate(ekCert)
+	if err != nil {
 		return nil, fmt.Errorf("parsing EK cert: %v", err)
 	}
+
+	// Initialize EK and compare public key to ekCert.PublicKey.
+	var ekh tpmutil.Handle
+	var ekPub crypto.PublicKey
+	if tmplIdx != 0 {
+		ekTemplate, err := tpm2.NVRead(rwc, tpmutil.Handle(tmplIdx))
+		if err != nil {
+			return nil, fmt.Errorf("reading EK template: %v", err)
+		}
+		ekh, ekPub, err = tpm2.CreatePrimaryRawTemplate(rwc, tpm2.HandleEndorsement, tpm2.PCRSelection{}, "", "", ekTemplate)
+		if err != nil {
+			return nil, fmt.Errorf("creating EK: %v", err)
+		}
+	} else {
+		// Default EK template defined in:
```

Comment:
I think it would be clearer to pull the default template out to the top as a var declaration, since it doesn't need to be embedded in the code here and is of independent interest if it's part of the spec.

Reply:
Done